File: src/main/app/com/leetcode/Strings/49_GroupAnagrams.java
package com.leetcode.Strings;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.HashMap;
import java.util.List;

class Solution_49 {
    public List<List<String>> groupAnagrams(String[] strs) {
        List<List<String>> list = new ArrayList<>();
        HashMap<String , ArrayList<String>> map = new HashMap<>();
        for(String s : strs){
            char[] ch = s.toCharArray();
            Arrays.sort(ch);
            String sorted_string = new String(ch);
            if(map.containsKey(sorted_string)){
                ArrayList<String> currlist = map.get(sorted_string);
                currlist.add(s);
                map.put(sorted_string,currlist);
            }
            else{
                ArrayList<String> li = new ArrayList();
                li.add(s);
                map.put(sorted_string,li);
            }
        }


        for(String s : map.keySet()){
            list.add(new ArrayList(map.get(s)));
        }

        return list;
    }
}

File: solutins-using-java-python/src/main/app/com/leetcode/Strings/49_GroupAnagrams.kt
package com.leetcode.Strings

import java.util.Arrays

class `49_GroupAnagrams` {
    fun groupAnagrams(strs: Array<String>): List<List<String>> {
        val map = mutableMapOf<String, MutableList<String>>()
        strs.forEach {
            val key = it.toCharArray().sorted().joinToString("")
            if (!map.containsKey(key)) map[key] = mutableListOf()
            map[key]?.add(it)
        }

        return map.values.toList()
    }
}